metals/src/main/scala/scala/meta/internal/builds/MavenBuildTool.scala

```diff
@@ -0,0 +1,63 @@
+package scala.meta.internal.builds
+
+import scala.meta.internal.metals.{MetalsServerConfig, UserConfiguration}
+import scala.meta.io.AbsolutePath
+import scala.tools.nsc.Properties
+import scala.meta.internal.metals.MetalsEnrichments._
+
+case class MavenBuildTool() extends BuildTool {
+
+  private lazy val embeddedMavenLauncher: AbsolutePath = {
+    val mavenWrapper =
+      if (Properties.isWin) "mvnw.cmd"
+      else "mvnw"
+    val out = BuildTool.copyFromResource(tempDir, mavenWrapper)
+    out.toFile.setExecutable(true)
+    Set(
+      s"maven-wrapper.jar",
+      "maven-wrapper.properties",
+      "MavenWrapperDownloader.java"
+    ).foreach { fileName =>
+      BuildTool.copyFromResource(
+        tempDir,
+        s"mvn/wrapper/$fileName",
+        Some(s".mvn/wrapper/$fileName")
+      )
+    }
+    AbsolutePath(out)
+  }
+
+  def args(
+      workspace: AbsolutePath,
+      userConfig: () => UserConfiguration,
+      config: MetalsServerConfig
+  ): List[String] = {
+    import scala.meta.internal.metals.BuildInfo
+    val command =
+      List(
+        s"ch.epfl.scala:maven-bloop_2.10:${BuildInfo.bloopVersion}:bloopInstall",
+        "-DdownloadSources=true"
+      )
+    userConfig().mavenScript match {
+      case Some(script) =>
+        script :: command
+      case None =>
+        embeddedMavenLauncher.toString() :: command
+    }
+  }
+
+  def digest(workspace: AbsolutePath): Option[String] =
+    MavenDigest.current(workspace)
+
+  def minimumVersion: String = "3.5.2"
+
+  def version: String = "3.6.1"
+
+  override def toString(): String = "maven"
```

Comment:
```suggestion
  override def toString(): String = "Maven"
```
?

Reply:
I think I need to rethink the tool name vs executable name in all of the build tools - I will create a follow up PR after mill is merged

Reply:
The message "New maven workspace detected" would read nicer with an uppercase `M`. Same for Mill.